packages/contracts/contracts/optimistic-ethereum/ovm/SafetyChecker.sol

```diff
@@ -85,72 +84,73 @@ contract SafetyChecker is ContractResolver {
                     // encountered a non-whitelisted opcode!
                     return false;
                 }
                 // append this opcode to a list of ops
-                ops[opIndex] = op;
-                // JUMPI
-                if (op == 0x57) {
-                    // We can now reach all JUMPDESTs
-                    seenJUMP = true;
-                // JUMP
-                } else if (op == 0x56) {
-                    // We can now reach all JUMPDESTs
-                    seenJUMP = true;
-                    // we are now inside unreachable code until we hit a JUMPDEST!
-                    insideUnreachableCode = true;
-                // STOP or REVERT or INVALID or RETURN (see safety checker docs in wiki for more info)
-                } else if (op == 0x00 || op == 0xfd || op == 0xfe || op == 0xf3) {
-                    // If we can't jump to JUMPDESTs, then all remaining bytecode is unreachable
-                    if (!seenJUMP) {
-                        return true;
-                    }
-                    // We are now inside unreachable code until we hit a JUMPDEST!
-                    insideUnreachableCode = true;
-                // CALL
-                } else if (op == 0xf1) {
-                    // Minimum 4 total ops:
-                    // 1. PUSH1 value
-                    // 2. PUSH20 execution manager address
-                    // 3. PUSH or DUP gas
-                    // 4. CALL
-
-                    if (opIndex < 3) {
-                        return false;
-                    }
-                    uint256 gasOp = ops[opIndex - 1];
-                    uint256 addressOp = ops[opIndex - 2];
-                    uint256 valueOp = ops[opIndex - 3];
-                    if (
-                        gasOp < 0x60 || // PUSHes are 0x60...0x7f
-                        gasOp > 0x8f || // DUPs are 0x80...0x8f
-                        addressOp != 0x73 || // address must be set with a PUSH20
-                        valueOp != 0x60 // value must be set with a PUSH1
-                    ) {
-                        return false;
-                    } else {
-                        uint256 pushedBytes;
-                        // gas is set with a PUSH##
-                        if (gasOp >= 0x60 && gasOp <= 0x7f) {
-                            pushedBytes = gasOp - 0x5f;
+                ops <<= 8;
+                ops |= op;
+                // [0x57, 0x56, 0x00, 0xfd, 0xfe, 0xf3, 0xf1] all have handlers
+                if (opBit & 0x600a00000000000000000000000000000000000000c000000000000000000001 != 0) {
+                    // JUMPI
+                    if (op == 0x57) {
+                        // We can now reach all JUMPDESTs
+                        seenJUMP = true;
+                    // JUMP
+                    } else if (op == 0x56) {
+                        // We can now reach all JUMPDESTs
+                        seenJUMP = true;
```

Comment:
now that I think about it, we can just assume that we've seen a `JUMP`. I doubt we'll ever have a case where we reach a halting opcode before seeing a `JUMP` (no need to update this in this PR, but curious about your thoughts)

Reply:
Meh, personally I'd remove this logic. It's safer the other way, never allow something you don't expect.